app/models/existing_role.rb

```diff
@@ -0,0 +1,34 @@
+class ExistingRole < ActiveRecord::Base
+  validates :name, presence: true
+  def self.update_resource(param)
+    id = param[:id]
+    name = param[:name]
+    role = where("id = ?", id).first
+    if role
+      role.update_attribute("name", name)
+      response = {
+        status: 200,
```

Comment:
The model should not be worrying about the http response code. It should just return the updated entity.